shopinvader_schema_sale/schemas/sale_order.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import datetime
+from typing import List
+
+from odoo.addons.extendable_fastapi import StrictExtendableBaseModel
+
+from .amount import SaleAmount
+from .invoicing import InvoicingInfo
+from .sale_order_line import SaleOrderLine
+from .shipping import ShippingInfo
+
+
+class BaseSale(StrictExtendableBaseModel):
+    id: int
+    state: str
+    name: str
+    client_order_ref: str | None = None
+    date_order: datetime
+    lines: List[SaleOrderLine]
+    amount: SaleAmount | None = None
+    # TODO discuss about this (should we keep the same schema for this field)
+    shipping: ShippingInfo | None = None
+    invoicing: InvoicingInfo | None = None
+    # TODO END
+    typology: str
```

Comment:
IMO, typology should go into the Sale schema. A cart will never have a typology != cart.

Reply:
It's something I would like to discuss. The aim of the typology was to differentiate a sale order between "stage" (a cart, a quotation, a sale order)

When I look at the code of the shopinvader_quotation module I realize that we have a lot of duplicated code in V1 (basically the endpoint "quotations" do the same thing as "sales" endpoint but it just apply a domain to have only quotation)

So maybe, it's will be simplier and more flexible to simply expose all sale order on the endpoint "sales" and be able to filter on the typology. So with the sale endpoint we can
- search for quotation
- search for sale
- search on cart (to support multi cart case)

not sure about this, but we should ask us the question to avoid duplicated code


In any case thanks for making a review on Sunday ;)